src/track/track.cpp

```diff
@@ -978,23 +1008,23 @@ bool Track::tryImportPendingBeatsMarkDirtyAndUnlock(
         return false;
     }
 
-    bool dirty = false;
-
+    bool modified = false;
+    // Both functions must be invoked even if one of them
+    // returns false!
     if (importPendingBeatsWhileLocked()) {
-        dirty = true;
+        modified = true;
     }
-
     if (compareAndSet(m_record.ptrBpmLocked(), lockBpmAfterSet)) {
-        dirty = true;
+        modified = true;
     }
-
-    if (!dirty) {
-        // Already set, nothing todo
-        pLock->unlock();
+    if (!modified) {
+        // Unmodified, nothing todo. Yet the internal dirty flag
+        // might have been set by a preceding operation!
+        markDirtyAndUnlock(pLock, false);
```

Comment:
This function call is misleading, because it does not mark the track dirty.
Even though it pretends to from the function name.
I think we should romove the result parameter and add a new funtion for this case .

Reply:
On further thought I think that simply unlocking and returning was sufficient here. If the dirty flag was set before a signal has already been sent.